libexec/fetch.ts

```diff
@@ -101,3 +111,45 @@ async function download({ dst, src }: { dst: Path, src: URL }) {
   }
   return dst
 }
+
+async function clone({ dst, src, ref }: { dst: Path, src: URL, ref: string }) {
+  const tmp = Path.mktemp({})
+  const proc = new Deno.Command("git", {
+    args: [
+      "clone",
+      "--quiet",
+      "--depth=1",
+      "--branch",
+      ref,
+      src.toString(),
+      tmp.string,
+    ],
+    // `git` uses stderr for... non errors, and --quiet
+    // doesn't touch them
+    stderr: "null",
+  })
+  const status = await proc.spawn().status
+  if (!status.success) {
+    console.error({ dst, src })
+    throw new Error(`git failed to clone ${src}`)
+  }
+
+  const proc2 = new Deno.Command("tar", {
```

Comment:
could use a comment. Took me a few minutes to realize you are creating a source tarball because presumably other parts of our infra require that to exist.

Can we not fix the other parts of the infra?

Reply:
Maybe, but should we? This seems smarter for general reproducibility. The only difference for git downloads is the initial download step.